examples/cra/src/hooks.ts

```diff
@@ -0,0 +1,372 @@
+import { useEffect, useState, useMemo, useRef, useCallback } from 'react'
+import { unstable_batchedUpdates } from 'react-dom'
+import Web3 from 'web3'
+import {
+    getProvider,
+    Provider,
+    isMetamaskProvider,
+    getProviderState,
+    createSubscriptions,
+    isWalletConnectProvider,
+    isMetamaskSubscriptions,
+    isWalletConnectSubscriptions,
+    WalletConnectInits,
+    WalletConnectProvider,
+} from '@gnosis.pm/dapp-ui'
+import { BlockHeader, Block } from 'web3-eth'
+
+// --------------------------------------------------
+
+// helper hook for if mounted check
+
+export const useIfMounted = () => {
```

Comment:
why not `useSafeState`

Reply:
This was made after I created `useIfMounted ` but before I thought of `useSafeState`